src/Data/API/BaseApiHelper.ts

```diff
@@ -9,12 +9,14 @@ import {
   ErrorWithHTTPCode,
 } from "@/Core";
 import { words } from "@/UI/words";
+import { getCookie } from "../Common/CookieHelper";
 import { BigIntJsonParser } from "./BigIntJsonParser";
 
 export class BaseApiHelper implements ApiHelper {
   jsonParser = new BigIntJsonParser();
   constructor(
     private readonly baseUrl: string = "",
+    private readonly shouldAuthLocally: boolean = false,
```

Comment:
why are you passing the shouldAuthLocally? I can't see it being used in the code below.